recipes/defaults.rb

```diff
@@ -10,7 +10,14 @@
 set :keep_releases,  5
 set :rake,           "govuk_setenv #{application} #{fetch(:rake, 'bundle exec rake')}"
 set :repo_name,      fetch(:repo_name, application).to_s # XXX: this must appear before the `require 'defaults' in recipe names
-set :repository,     "#{ENV.fetch('GIT_ORIGIN_PREFIX', 'git@github.com:alphagov')}/#{repo_name}.git"
+
+set :repo_prefix,    ENV.fetch('GIT_ORIGIN_PREFIX', 'git@github.com:alphagov')
```

Comment:
Github is pretty permissive, it doesn't require the `.git` suffix. So I think you might be able to remove this logic and just never add `.git`?